### PR TITLE
solution22 is completed

### PR DESCRIPTION
**Pull request by Jayshgandhi**

### CODE REVIEW

File: Exercises/Excercise22/solution22.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                Collectors.partitioningBy(n -> n.getStudent_mark() > passMarkThreshold));
+        return map;
+    }
+}
+
+public class solution22 {
+    public static void main(String[] args) {
+        List<Student> sd = new ArrayList<Student>();
+        List<Integer> in =new ArrayList<>();
+        Name_generator ex = new Name_generator();
```
Comment (fr-judson) [inline Exercises/Excercise22/solution22.java:102]: Include Name_generator code

File: Exercises/Excercise22/solution22.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+        List<String> res = d.stream().filter(v ->v.student_mark>passMarkThreshold).map(v -> v.getStudent_name()).collect(Collectors.toList());
+        return res;
+
+    }
+    public Optional<Student> firstRankStudent(List<Student> r){
+        Optional<Student> fr = r.stream().max((o1, o2) -> o1.student_mark.compareTo(o2.student_mark));
+        return fr;
+    }
+    public List<String> numberOfFailures(List<Student> d,Integer passMarkThreshold){
```
Comment (fr-judson) [inline Exercises/Excercise22/solution22.java]: Expected result is in integer of number of students failed on the exam

**Review by fr-judson — CHANGES_REQUESTED** [review]
Please fix these changes

File: Exercises/Excercise22/solution22.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        return fr;
+    }
+    public Integer numberOfFailures(List<Student> d,Integer passMarkThreshold){
+
+        Integer res = Math.toIntExact(d.stream().filter(s -> s.student_mark < passMarkThreshold).count());
+        return res;
+
+    }
+    public void printlastFiveStudents(List<Student> s){
+        List<Student> gt =s.subList(Math.max(s.size()-5,0), s.size());
```
Comment (fr-judson) [inline Exercises/Excercise22/solution22.java]: Use stream to get last five students and print the students also using stream

**Review by fr-judson — CHANGES_REQUESTED** [review]
LGTM

File: Exercises/Excercise22/solution22.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    }
+    public Integer numberOfFailures(List<Student> d,Integer passMarkThreshold){
+
+        Integer res = Math.toIntExact(d.stream().filter(s -> s.student_mark < passMarkThreshold).count());
+        return res;
+
+    }
+    public void printlastFiveStudents(List<Student> s){
+        int count= (int) s.stream().count();
+        List<Student> t = s.stream().skip(count-5).collect(Collectors.toList());
```
Comment (hockeyjudson) [inline Exercises/Excercise22/solution22.java]: Use the same stream line to print

**Review by hockeyjudson — CHANGES_REQUESTED** [review]

File: Exercises/Excercise22/solution22.java (hunk 4 above)
Comment (hockeyjudson) [reply Exercises/Excercise22/solution22.java]: Also write code to handle the case for the number of elements less than five use if else

Comment (hockeyjudson) [reply Exercises/Excercise22/solution22.java]: Stream print https://www.geeksforgeeks.org/how-to-print-elements-of-a-stream-in-java-8/

**Review by hockeyjudson — APPROVED** [review]
LGTM

**Pull request merged**